### PR TITLE
Unit Tests

### PR DESCRIPTION
**Pull request by CorruptComputer**
Very much a work in progress.

Status:
- [ ] Models/Api/Server.cs (in progress)
- [ ] Models/Api/ServerList.cs (in progress)
- [ ] Models/ConfigFile/Config.cs (in progress)
- [ ] Models/ConfigFile/HubClientConfig.cs (in progress)
- [ ] Models/ConfigFile/Preferences.cs (in progress)
- [ ] Models/Download.cs (in progress)
- [x] Models/Installation.cs
- [ ] Services/AuthService.cs (in progress)
- [ ] Services/DownloadService.cs
- [ ] Services/InstallationService.cs
- [ ] Services/ServerService.cs
- [ ] Services/StateService.cs
- [x] ValueConverters/HumanizerValueConverter.cs

### CODE REVIEW

**Comment by CorruptComputer** [comment]
I haven't forgot about this, just been busier than expected. Have Unit Tests started for the classes in the Models directory locally that are not finished. Hopefully will have time this week to work on those and get them pushed up

File: UnitystationLauncher.UnitTests/Models/InstallationTests.cs [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    [Fact]
+    public void GetBuildVersion_ShouldHandleValidInputs()
+    {
+        string buildVersionTestDirectory = "BuildVersionTest23020422";
+        int actual = Installation.GetBuildVersion(buildVersionTestDirectory);
+        actual.Should().Be(23020422);
+    }
+    #endregion
+
+    #region Installation.MakeExecutableExecutable
```
Comment (corp-0) [inline UnitystationLauncher.UnitTests/Models/InstallationTests.cs:122]: why so many regions?

Comment (CorruptComputer) [reply UnitystationLauncher.UnitTests/Models/InstallationTests.cs:122]: Region for each method being tested, makes it nice when you're trying to look at the tests for one specific method if it is failing: 
image

Being able to just collapse all the rest makes it easier to read through imo

**Comment by CorruptComputer** [comment]
Going to close this for now, will continue on it after the 931 release. Will probably split up the tests into separate PRs to reduce the code review burden. Plan is to start with the services (skip AuthService as it is going to be removed) since those are fairly self contained.

**Pull request closed without merging**